vcd/resource_vcd_multisite_site_association.go

```diff
@@ -0,0 +1,206 @@
+		associationData, err := govcd.RawDataToStructuredXml[types.SiteAssociationMember](associatedRawData)
+		if err != nil {
+			return diag.Errorf("error decoding data from 'association_data_file' %s: %s", associationDataFile, err)
+		}
+		associatedSiteId = associationData.SiteID
+	}
+	if associatedSiteId == "" {
+		return diag.Errorf("no site ID found in either d.Id() or 'associated_site_id' field")
+	}
+	// Note: the data retrieved by the operation below only exists if an association has already been established.
+	// The existence of an XML file containing an associated site is just a convenient way of retrieving the site ID.
+	associationData, err := client.Client.GetSiteAssociationBySiteId(associatedSiteId)
+	if err != nil {
+		if origin == "datasource" {
+			return diag.Errorf("error retrieving association data for site ID '%s': %s", associatedSiteId,
+				fmt.Errorf("%s: %s", govcd.ErrorEntityNotFound, err))
+		}
+		d.SetId("")
+		return nil
+	}
+	dSet(d, "associated_site_id", associationData.SiteID)
+	dSet(d, "associated_site_name", associationData.SiteName)
+	dSet(d, "associated_site_href", associationData.Href)
+	dSet(d, "status", associationData.Status)
+	d.SetId(associationData.SiteID)
+
+	return nil
+}
+
+// resourceVcdSiteAssociationUpdate will only update "connection_timeout_mins"
+func resourceVcdSiteAssociationUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*VCDClient)
+	associatedSiteId := d.Id()
+	connectionCheckMinutes := d.Get("connection_timeout_mins").(int)
+	if d.HasChange("connection_timeout_mins") && connectionCheckMinutes > 0 {
+		status, elapsed, err := client.Client.CheckSiteAssociation(associatedSiteId, time.Minute*time.Duration(connectionCheckMinutes))
+		if err != nil {
+			return diag.Errorf("error checking for site connection after %s - detected status '%s': %s", elapsed, status, err)
+		}
+	}
+	return resourceVcdSiteAssociationRead(ctx, d, meta)
+}
+
+func resourceVcdSiteAssociationDelete(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*VCDClient)
+	associatedSiteId := d.Id()
+	if associatedSiteId == "" {
+		associatedSiteId = d.Get("associated_site_id").(string)
+	}
+	if associatedSiteId == "" {
+		return diag.Errorf("no site ID found in either d.Id() or 'associated_site_id' field")
+	}
+	associationData, err := client.Client.GetSiteAssociationBySiteId(associatedSiteId)
+	if err != nil {
+		return diag.Errorf("error retrieving association data for site ID '%s': %s", associatedSiteId, err)
+	}
+	err = client.Client.RemoveSiteAssociation(associationData.Href)
+	if err != nil {
+		return diag.Errorf("error removing site '%s': %s", associationData.SiteName, err)
+	}
+	return nil
+}
+
+func resourceVcdSiteAssociationImport(ctx context.Context, d *schema.ResourceData, meta interface{}) ([]*schema.ResourceData, error) {
+	client := meta.(*VCDClient)
+
+	associatedSiteId := d.Id()
```

Comment:
Should we explain this in the docs? (I don't see a section for Import).

Reply:
Added